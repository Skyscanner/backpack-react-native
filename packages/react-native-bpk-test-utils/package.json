{
  "name": "react-native-bpk-test-utils",
  "version": "0.0.5",
  "private": true,
  "main": "index.js",
  "description": "Backpack React Native test utlities.",
  "repository": {
    "type": "git",
    "url": "git@github.com:Skyscanner/backpack.git"
  },
  "author": "Backpack Design System <backpack@skyscanner.net>",
  "license": "Apache-2.0",
  "peerDependencies": {
    "react": "^16.8.3",
    "react-native": ">= 0.59.0",
    "react-native-bpk-appearance": ">= 1.3.1"
  }
}
