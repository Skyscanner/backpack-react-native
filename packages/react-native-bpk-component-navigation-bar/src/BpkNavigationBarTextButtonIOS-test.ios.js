/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016-2019 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* @flow */

import React from 'react';
import renderer from 'react-test-renderer';
import { describeEachColorScheme } from 'react-native-bpk-test-utils';

import BpkNavigationBarTextButtonIOS, {
  BUTTON_TYPES,
} from './BpkNavigationBarTextButtonIOS.ios';

jest.mock('bpk-tokens/tokens/base.react.native', () =>
  jest.requireActual('bpk-tokens/tokens/base.react.native.android.js'),
);

describe('iOS', () => {
  describe('BpkNavigationBarTextButtonIOS', () => {
    describeEachColorScheme(BpkNavigationBarTextButtonIOS, WithColorScheme => {
      it('should render correctly', () => {
        const tree = renderer
          .create(
            <WithColorScheme
              title="Edit"
              leading={false}
              onPress={jest.fn()}
            />,
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should render correctly leading configuration', () => {
        const tree = renderer
          .create(
            <WithColorScheme title="Cancel" leading onPress={jest.fn()} />,
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should render correctly emphasize', () => {
        const tree = renderer
          .create(
            <WithColorScheme
              title="Done"
              emphasize
              leading={false}
              onPress={jest.fn()}
            />,
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should render correctly disabled', () => {
        const tree = renderer
          .create(
            <WithColorScheme
              title="Done"
              leading={false}
              onPress={jest.fn()}
              disabled
            />,
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should render correctly with type="primary"', () => {
        const tree = renderer
          .create(
            <WithColorScheme
              title="Done"
              leading={false}
              onPress={jest.fn()}
              type={BUTTON_TYPES.primary}
            />,
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should render correctly with type="primary" and disabled', () => {
        const tree = renderer
          .create(
            <WithColorScheme
              title="Done"
              leading={false}
              onPress={jest.fn()}
              type={BUTTON_TYPES.primary}
              disabled
            />,
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should respect "tintColor" over "type"', () => {
        const tree = renderer
          .create(
            <WithColorScheme
              title="Done"
              leading={false}
              onPress={jest.fn()}
              type={BUTTON_TYPES.primary}
              tintColor="red"
            />,
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should respect "disabledTintColor" over "type"', () => {
        const tree = renderer
          .create(
            <WithColorScheme
              title="Done"
              leading={false}
              onPress={jest.fn()}
              disabledTintColor="red"
              type={BUTTON_TYPES.primary}
              disabled
            />,
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should respect "primaryTintColor" over "tintColor" if type="primary"', () => {
        const tree = renderer
          .create(
            <WithColorScheme
              title="Done"
              leading={false}
              onPress={jest.fn()}
              type={BUTTON_TYPES.primary}
              tintColor="red"
              disabledTintColor="yellow"
              primaryTintColor="blue"
            />,
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should respect "tintColor" over "primaryTintColor" if type="default"', () => {
        const tree = renderer
          .create(
            <WithColorScheme
              title="Done"
              leading={false}
              onPress={jest.fn()}
              tintColor="red"
              disabledTintColor="yellow"
              primaryTintColor="blue"
            />,
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
