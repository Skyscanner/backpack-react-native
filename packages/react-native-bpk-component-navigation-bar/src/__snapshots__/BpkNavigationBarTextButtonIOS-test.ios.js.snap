// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iOS BpkNavigationBarTextButtonIOS should render correctly 1`] = `
<View
  accessibilityLabel="Edit"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginEnd": 8,
      "opacity": 1,
      "zIndex": 2,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgb(82, 76, 97)",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.12,
        },
        Object {},
        Array [
          Object {
            "color": "rgb(82, 76, 97)",
          },
        ],
      ]
    }
  >
    Edit
  </Text>
</View>
`;

exports[`iOS BpkNavigationBarTextButtonIOS should render correctly disabled 1`] = `
<View
  accessibilityLabel="Done"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginEnd": 8,
      "opacity": 1,
      "zIndex": 2,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgb(82, 76, 97)",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.12,
        },
        Object {},
        Array [
          Object {
            "color": "rgb(178, 174, 189)",
          },
        ],
      ]
    }
  >
    Done
  </Text>
</View>
`;

exports[`iOS BpkNavigationBarTextButtonIOS should render correctly emphasize 1`] = `
<View
  accessibilityLabel="Done"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginEnd": 8,
      "opacity": 1,
      "zIndex": 2,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgb(82, 76, 97)",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.12,
        },
        Object {
          "fontWeight": "600",
        },
        Array [
          Object {
            "color": "rgb(82, 76, 97)",
          },
        ],
      ]
    }
  >
    Done
  </Text>
</View>
`;

exports[`iOS BpkNavigationBarTextButtonIOS should render correctly leading configuration 1`] = `
<View
  accessibilityLabel="Cancel"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginStart": 8,
      "opacity": 1,
      "zIndex": 2,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgb(82, 76, 97)",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.12,
        },
        Object {},
        Array [
          Object {
            "color": "rgb(82, 76, 97)",
          },
        ],
      ]
    }
  >
    Cancel
  </Text>
</View>
`;

exports[`iOS BpkNavigationBarTextButtonIOS should render correctly with type="primary" 1`] = `
<View
  accessibilityLabel="Done"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginEnd": 8,
      "opacity": 1,
      "zIndex": 2,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgb(82, 76, 97)",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.12,
        },
        Object {},
        Array [
          Object {
            "color": "rgb(0, 178, 214)",
          },
        ],
      ]
    }
  >
    Done
  </Text>
</View>
`;

exports[`iOS BpkNavigationBarTextButtonIOS should render correctly with type="primary" and disabled 1`] = `
<View
  accessibilityLabel="Done"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginEnd": 8,
      "opacity": 1,
      "zIndex": 2,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgb(82, 76, 97)",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.12,
        },
        Object {},
        Array [
          Object {
            "color": "rgb(178, 174, 189)",
          },
        ],
      ]
    }
  >
    Done
  </Text>
</View>
`;

exports[`iOS BpkNavigationBarTextButtonIOS should respect "disabledTintColor" over "type" 1`] = `
<View
  accessibilityLabel="Done"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginEnd": 8,
      "opacity": 1,
      "zIndex": 2,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgb(82, 76, 97)",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.12,
        },
        Object {},
        Array [
          Object {
            "color": "red",
          },
        ],
      ]
    }
  >
    Done
  </Text>
</View>
`;

exports[`iOS BpkNavigationBarTextButtonIOS should respect "primaryTintColor" over "tintColor" if type="primary" 1`] = `
<View
  accessibilityLabel="Done"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginEnd": 8,
      "opacity": 1,
      "zIndex": 2,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgb(82, 76, 97)",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.12,
        },
        Object {},
        Array [
          Object {
            "color": "blue",
          },
        ],
      ]
    }
  >
    Done
  </Text>
</View>
`;

exports[`iOS BpkNavigationBarTextButtonIOS should respect "tintColor" over "primaryTintColor" if type="default" 1`] = `
<View
  accessibilityLabel="Done"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginEnd": 8,
      "opacity": 1,
      "zIndex": 2,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgb(82, 76, 97)",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.12,
        },
        Object {},
        Array [
          Object {
            "color": "red",
          },
        ],
      ]
    }
  >
    Done
  </Text>
</View>
`;

exports[`iOS BpkNavigationBarTextButtonIOS should respect "tintColor" over "type" 1`] = `
<View
  accessibilityLabel="Done"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginEnd": 8,
      "opacity": 1,
      "zIndex": 2,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgb(82, 76, 97)",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.12,
        },
        Object {},
        Array [
          Object {
            "color": "red",
          },
        ],
      ]
    }
  >
    Done
  </Text>
</View>
`;
